src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts

```diff
@@ -360,4 +360,22 @@ export class StudentExamsComponent implements OnInit {
         // TODO: we should try to use the artemis date pipe here
         return date ? moment(date).format(defaultLongDateTimeFormat) : '';
     }
+
+    /**
+     * Shows the translated error message if an error key is available in the error response. Otherwise it defaults to the generic alert.
+     * @param translationString the string identifier in the translation service for the text. This is ignored if the response does not contain an error message or error key.
+     * @param err the error response
+     */
+    private handleError(translationString: string, err: HttpErrorResponse) {
+        let errorDetail = err?.error?.message;
+        if (err?.error && err.error.errorKey) {
+            errorDetail = this.translateService.instant(err.error.errorKey);
+        }
```

Comment:
```suggestion
        let errorDetail;
        if (err?.error && err.error.errorKey) {
            errorDetail = this.translateService.instant(err.error.errorKey);
        } else {
            errorDetail = err?.error?.message;
        }
```
Don't know how much of an improvement this is to you, but as if the errorKey exists, we overwrite the errorDetail anyway, we could throw this into an else statement to one assignment

Reply:
Yeah I agree. Performance-wise it won't make much difference, but I think your suggestion is more readable. I will change it